python/paddle/nn/functional/pooling.py

```diff
@@ -13,9 +13,248 @@
 # limitations under the License.
 
 # TODO: define pooling functions
+import paddle
 from ...fluid.layers import pool2d  #DEFINE_ALIAS
 from ...fluid.layers import pool3d  #DEFINE_ALIAS
 from ...fluid.layers import adaptive_pool2d  #DEFINE_ALIAS
 from ...fluid.layers import adaptive_pool3d  #DEFINE_ALIAS
+from ...fluid.data_feeder import convert_dtype, check_variable_and_dtype, check_type, check_dtype
+from ...fluid.layers import utils
+from ...fluid.layer_helper import LayerHelper
 
-__all__ = ['pool2d', 'pool3d', 'adaptive_pool2d', 'adaptive_pool3d']
+__all__ = [
+    'pool2d', 'pool3d', 'adaptive_pool2d', 'adaptive_pool3d',
+    'adaptive_avg_pool2d', 'adaptive_avg_pool3d'
+]
+
+
+def adaptive_avg_pool2d(x, output_size, data_format='NCHW', name=None):
+    """
+
+    This operation applies 2D adaptive avg pooling on input tensor. The h and w dimensions
```

Comment:
refer to layer docstring?

Reply:
done, thanks

Reply:
This API description(formula) is already mentioned in layer